pkg/cli/root_test.go

```diff
@@ -1018,11 +1018,56 @@ func TestVerify(t *testing.T) {
 	})
 }
 
+func TestAutoPortSelection(t *testing.T) {
+	Convey("Run server without specifying a port (port 0)", t, func() {
+		logFile, err := os.CreateTemp("", "zot-log*.txt")
+		logPath := logFile.Name()
+		So(err, ShouldBeNil)
+
+		content := fmt.Sprintf(`{"distSpecVersion": "1.0.0", "storage": {"rootDirectory": "/tmp/zot"},
+		"http": {"address": "127.0.0.1", "port": "0"},
+		"log": {"level": "debug", "output": "%s"}}`, logFile.Name())
+
+		cfgfile, err := os.CreateTemp("", "zot-test*.json")
+		So(err, ShouldBeNil)
+
+		defer os.Remove(cfgfile.Name()) // clean up
+
+		_, err = cfgfile.Write([]byte(content))
+		So(err, ShouldBeNil)
+
+		os.Args = []string{"cli_test", "serve", cfgfile.Name()}
+		go func() {
+			err = cli.NewServerRootCmd().Execute()
+			So(err, ShouldBeNil)
+		}()
+
+		_, err = ReadLogFileAndSearchString(logPath, "CVE config not provided, skipping CVE update", 90*time.Second)
+		if err != nil {
+			panic(err)
+		}
+
+		data, err := os.ReadFile(logFile.Name())
+		So(err, ShouldBeNil)
+		So(string(data), ShouldContainSubstring, "\"Port\":")
+	})
+}
+
 func TestLoadConfig(t *testing.T) {
```

Comment:
Is there a port-related reason this test needed to change?

Reply:
I changed it because now the port is set when the config is verified (the port is now set in roots.go), and in the old version of this test the port on the controller would remain 0 because no config was loaded, and the verification wouldn't take place, thus leaving the port with value 0.